thavalon-webapp/src/components/gameComponents/missionResults.tsx

```diff
@@ -1,5 +1,176 @@
+        {
+            missionPlayers: ["Mission 1"],
+            missionStatus: MissionStatus.Pending,
+            missionNumber: 1,
+        },
+        {
+            missionPlayers: ["Mission 2"],
+            missionStatus: MissionStatus.Pending,
+            missionNumber: 2,
+        },
+        {
+            missionPlayers: ["Mission 3"],
+            missionStatus: MissionStatus.Pending,
+            missionNumber: 3,
+        },
+        {
+            missionPlayers: ["Mission 4"],
+            missionStatus: MissionStatus.Pending,
+            missionNumber: 4,
+        },
+        {
+            missionPlayers: ["Mission 5"],
+            missionStatus: MissionStatus.Pending,
+            missionNumber: 5,
+        }
+    ])
+
+    // useEffect handles componentDidMount and componentWillUnmount steps.
+    useEffect(() => {
+        // On mount, get the connection instance and set up event handlers.
+        // Then, get the player list.
+        const connection = GameSocket.getInstance();
+        connection.onGameEvent.subscribe(handleMessage);
+
+        // On unmount, unsubscribe our event handlers.
+        return () => {
+            connection.onGameEvent.unsubscribe(handleMessage);
+        }
+    }, [missionProps]);
+
+    /**
+     * Handles any game messages that come from the server. Currently supports
+     * MissionGoing and MissionResults game messages.
+     * @param message An incoming GameMessage from the server.
+     */
+    function handleGameMessage(message: GameMessage) {
+        switch (message.messageType) {
+            case GameMessageType.MissionGoing: {
+                const missionGoingData = (message.data as MissionGoingMessage);
+                const newArr = new Array(...missionProps);
+                newArr[missionGoingData.mission - 1] = {
+                    missionPlayers: Array.from(missionGoingData.players),
+                    missionStatus: MissionStatus.Going,
+                    missionNumber: missionGoingData.mission,
+                }
+                setMissionProps(newArr);
+                break;
+            }
+            case GameMessageType.MissionResults: {
+                let missionResultsData = (message.data as MissionResultsMessage);
+                let newArr = new Array(...missionProps);
+                newArr[missionResultsData.mission - 1] = {
+                    missionPlayers: missionProps[missionResultsData.mission - 1].missionPlayers,
+                    missionStatus: missionResultsData.passed ? MissionStatus.Passed : MissionStatus.Failed,
+                    missionNumber: missionResultsData.mission,
+                    passes: missionResultsData.successes,
+                    fails: missionResultsData.fails,
+                    reverses: missionResultsData.reverses,
+                    questing_beasts: missionResultsData.questing_beasts,
+                }
+                setMissionProps(newArr);
```

Comment:
Question maybe for @bnavetta, does agravaine declaring resend the mission results, or does that need to be handled here?

Reply:
No, instead it sends an `AgravaineDeclaration` message. I think we need to handle that here by updating the mission status